### DISCUSSION TITLE
Supabase Hiccups on Mobile (React-Native)

### DISCUSSION THREAD

**Discussion by madprogramer** [supabase/supabase#1293] [Questions] [answered]
Hi there!

Our team has been using Supabase for the past few months to build a crowd-sourced news project [called Crowdie](https://github.com/Crowdie-Project/Crowdie). We were using React Native and things had gone so far, so good. _That is until we began testing authentication on mobile._

We're getting a ton of un-definition errors:
```bash
not implemented
at node_modules\react-native\Libraries\Blob\URL.js:86:10 in URLSearchParams#set
at node_modules\@supabase\postgrest-js\dist\main\lib\PostgrestQueryBuilder.js:26:8 in select
...
```

It would seem we were not alone, as this fellow called [Justin Noel](https://justinnoel.dev/2020/12/08/react-native-urlsearchparams-error-not-implemented/) wrote a blog post about this last year.

> The problem stems from the Supabase PostgREST-js library using the URLSearchParams Web API to convert this Supabase query into a URL.
> ```js
>   await supabase.from<todos>('todos').select().order('id')
>   https://[my-project-id].supabase.co/rest/v1/todos?select=*&order=id.asc.nullslast
> ```
> With just a little bit of searching, I found that the React Native react-native/Libraries/Blob/URL.js doesn't import all the Web API functionality due to size concerns.

Unfortunately, [Justin (and Ahmed Tarek's) solution](https://github.com/facebook/react-native/issues/23922#issuecomment-648096619) of installing `react-native-url-polyfill/auto` was not enough to fix our problems. 

If you have used Supabase for a React Native project before, have you ever dealt with a similar series of errors before?

Thanks in advance :}

**Comment by soedirgo** [comment]
We had another user facing this exact same issue (see https://github.com/supabase/postgrest-js/issues/154), and the polyfill solution above fixed it. Re: the `not implemented` error above, you shouldn't getting those anymore once you import the polyfill, since AFAICT it only comes from the guilty `URL` implementation (plus from `metro`, but it doesn't look like it's relevant):
```
❯ rg -uuu '\'not implemented\'' .
./my-repro-app/node_modules/metro/src/ModuleGraph/node-haste/node-haste.js.flow
50:    throw new Error('not implemented');
53:    throw new Error('not implemented');

./my-repro-app/node_modules/react-native/Libraries/Blob/URL.js
70:    throw new Error('not implemented');
74:    throw new Error('not implemented');
78:    throw new Error('not implemented');
82:    throw new Error('not implemented');
86:    throw new Error('not implemented');
90:    throw new Error('not implemented');
161:    throw new Error('not implemented');
165:    throw new Error('not implemented');
169:    throw new Error('not implemented');
177:    throw new Error('not implemented');
181:    throw new Error('not implemented');
185:    throw new Error('not implemented');
189:    throw new Error('not implemented');
193:    throw new Error('not implemented');
197:    throw new Error('not implemented');
220:    throw new Error('not implemented');
```

**Comment by madprogramer** [comment, marked as answer]
@kiwicopple actually had the [missing piece](https://www.gitmemory.com/issue/supabase/supabase-js/129/778872778). Expo requires an async-storage, go figure! Worked like a charm, afterwards.
